packages/eslint-plugin/src/rules/consistent-generic-constructors.ts

```diff
@@ -33,6 +33,30 @@ export default createRule<Options, MessageIds>({
   },
   defaultOptions: ['constructor'],
   create(context, [mode]) {
+    const isTypedArrayReference = (typeNode: TSESTree.TypeNode) => {
```

Comment:
[Style] It's probably optimized away by engines, but to make the code more clear, a few suggestions:

* Move the `typedArrayNames` to a single declared array outside the `createRule`, so it doesn't look like it's declared differently on each check
* Use a `Set` for O(1) `.has`, which is really what's being done
* There only needs to be one `return`

I.e. let's turn this function into:

```ts
function isTypedArrayReference(typeNode: TSESTree.TypeNode) {
  return (
    typeNode.type === AST_NODE_TYPES.TSTypeReference &&
    typeNode.typeName.type === AST_NODE_TYPES.Identifier &&
    typedArrayNames.has(typeNode.typeName.name)
  );
}
```